planseqlearn/psl/mopa_mp_env.py

```diff
@@ -426,47 +425,23 @@ def __init__(
             self.sim.model.geom_rgba[idx].copy() for idx in self.robot_geom_ids
         ]
         self.retry = False
-        self.text_plan = [("null", "null")]
 
-    def check_robot_collision(self, **kwargs):
-        return check_collisions(
-            self, self.allowed_collision_pairs, self.env_name, **args, **kwargs
-        )
-
-    def set_robot_colors(self, colors):
-        if type(colors) is np.ndarray:
-            colors = [colors] * len(self.robot_geom_ids)
-        for idx, geom_id in enumerate(self.robot_geom_ids):
-            self.sim.model.geom_rgba[geom_id] = colors[idx]
-        self.sim.forward()
-
-    def reset_robot_colors(self):
-        self.set_robot_colors(self.original_colors)
-        self.sim.forward()
-
-    def get_object_pose(self, **kwargs):
-        if self.env_name == "SawyerLiftObstacle-v0" or self.env_name == "SawyerLift-v0":
-            start = self.sim.model.body_jntadr[self.sim.model.body_name2id("cube")]
-            object_pos = self.sim.model.body_jntadr[self.sim.model.body_name2id("cube")]
-            object_pos = self.sim.data.qpos[start : start + 3].copy()
-            object_quat = self.sim.data.qpos[start + 3 : start + 7].copy()
-            object_quat = convert_quat(object_quat, to="xyzw")
-        elif self.env_name == "SawyerAssemblyObstacle-v0":
-            object_pos = get_site_pose(self._wrapped_env, "hole")[0]
-            object_quat = None
-        elif self.env_name == "SawyerPushObstacle-v0":
-            object_pos = np.array(
-                self._wrapped_env.sim.data.body_xpos[self._wrapped_env.cube_body_id]
-            )
-            object_quat = None
-        else:
-            raise NotImplementedError
-        return object_pos, object_quat
-
-    def get_object_pose_mp(self, **kwargs):
-        if self.env_name == "SawyerLiftObstacle-v0" or self.env_name == "SawyerLift-v0":
-            object_pos, object_quat = self.get_object_pose()
-            if self.use_vision_pose_estimation:
+    def get_hardcoded_text_plan(self):
+        if self.env_name == "SawyerAssemblyObstacle-v0":
+            return [("empty hole", "place")]
+        if self.env_name == "SawyerLiftObstacle-v0":
+            return [("red can", "grasp")]
+        if self.env_name == "SawyerPushObstacle-v0":
+            return [("red cube", "grasp")]
+    
+    def get_curr_postcondition():
+        pass 
+    
+    def named_get_object_pose_mp(self, obj_name):
+        if "can" in obj_name:
```

Comment:
it seems like self.sam_object_pose[obj_name] is repeated. can you re-write this to instead be: 
if self.use_sam_seg:
...
else:
 if 'can':
...
elif 'hole':
...
so on